Python/getargs.c

```diff
@@ -480,51 +481,97 @@ converttuple(PyObject *arg, const char **p_format, va_list *p_va, int flags,
         }
         else if (c == ':' || c == ';' || c == '\0')
             break;
-        else if (level == 0 && Py_ISALPHA(c) && c != 'e')
-            n++;
+        else {
+            if (level == 0 && Py_ISALPHA(c))
+                n++;
+            if (c == 'e' && (*format == 's' || *format == 't')) {
+                format++;
+                continue;
+            }
+            if (!mustbetuple) {
+                switch (c) {
+                    case 'y':
+                    case 's':
+                    case 'z':
+                        if (*format != '*') {
+                            mustbetuple = 1;
+                        }
+                        break;
+                    case 'S':
+                    case 'Y':
+                    case 'U':
+                        mustbetuple = 1;
+                        break;
+                    case 'O':
+                        if (*format != '&') {
+                            mustbetuple = 1;
+                        }
+                        break;
+                }
+            }
+        }
     }
 
-    if (!PySequence_Check(arg) || PyBytes_Check(arg)) {
+    if (PyTuple_Check(arg)) {
+        Py_INCREF(arg);
+    }
```

Comment:
We can easily avoid the unneeded incref/decref if we refactor out the `convertitem` loop. I'll make a suggestion to your fork for your consideration.

Reply:
Suggestion opened:

- https://github.com/serhiy-storchaka/cpython/pull/23

Reply:
> So I left the simpler code.

However, the "simpler" code is more complex with regards to reference counting. IMO, keeping the ref counting simple is worth it; ref count bugs are hard to catch. But it is your call. BTW, I addressed your remarks on your fork.

Reply:
The code now uses a separate flag to avoid incref/decref.